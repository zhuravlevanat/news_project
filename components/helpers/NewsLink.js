import Link from "next/link";
import React from 'react';
import ListItem from '@material-ui/core/ListItem';
import MyLink from '@material-ui/core/Link';
import { makeStyles } from '@material-ui/core/styles';


const useStyles = makeStyles( theme => ({
    root: {
        cursor: 'pointer',
        color: theme.palette.info.main,
        fontWeight: 'bold',
        textTransform: 'capitalize',
        fontSize: '1.5rem',
    },
}))


const ListItemLink = props => {
    const classes = useStyles();
    return (
         <ListItem>
           <Link href={`/news?category=${props.title}`}>
                <MyLink classes={{root: classes.root}}>{props.title}</MyLink>
            </Link>
        </ListItem>
    )
};


export default ListItemLink;