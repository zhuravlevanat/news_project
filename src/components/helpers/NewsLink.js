import Link from "next/link";
import React from 'react';
import ListItem from '@material-ui/core/ListItem';
import MyLink from '@material-ui/core/Link';
import { useStyles } from "./NewsLinkStyles";

const ListItemLink = ({ title }) => {
    const classes = useStyles();
    return (
        <ListItem>
            <Link href={`/news?category=${title}`}>
                <MyLink className={classes.root}>{title}</MyLink>
            </Link>
        </ListItem>
    );
};

export default ListItemLink;
