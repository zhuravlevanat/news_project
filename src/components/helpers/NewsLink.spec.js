import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import NewsLink from "./NewsLink";

Enzyme.configure({ adapter: new Adapter() });

describe('NewsLink', () => {
    describe('NewsLink wrapper', () => {
        const props = {
            title: 'sport',
        }
        const link = shallow(<NewsLink {...props}/>);
        it('renders without crashing', () => {
            expect(toJson(link)).toMatchSnapshot();
        });
        it('renders 1 link wrapper', () => {
            expect(link.find('WithStyles(ForwardRef(ListItem))')).toHaveLength(1);
        });
    });
    describe('NewLink title', () => {
        const nextProps = {
            title: 'film',
        };
        const link = mount(<NewsLink {...nextProps}/>);
        it('NewsLink contains title', () => {
            expect(link.find('Link').text()).toEqual(nextProps.title);
        });
    });
});
