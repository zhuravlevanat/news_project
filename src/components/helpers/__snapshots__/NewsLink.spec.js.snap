// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsLink NewsLink wrapper renders without crashing 1`] = `
<WithStyles(ForwardRef(ListItem))>
  <Link
    href="/news?category=sport"
  >
    <WithStyles(ForwardRef(Link))
      className="makeStyles-root-1"
    >
      sport
    </WithStyles(ForwardRef(Link))>
  </Link>
</WithStyles(ForwardRef(ListItem))>
`;
